src/api/account.ts

```diff
@@ -0,0 +1,137 @@
+import { query } from '../db';
+import { cache } from '../cache';
+import {
+  banPlayerByAccIdQuery,
+  unbanPlayerByAccIdQuery,
+  getAccIdByUserNameQuery,
+  deleteSessionByAccIdQuery,
+} from './account.queries';
+import type { AccountId } from './account.types';
+import type { UpdateQuery } from '../types';
+
+/**
+ * Gets Account Id by Username
+ *
+ * @param {string} userName The user name / login of the account
+ * @returns {number} - if -1 then no Account Id Found
+ */
+export const getAccountIdByUserName = async (userName: string): Promise<number> => {
+  return cache.get(
+    {
+      key: `getAccountByUserName_${userName}`,
+      ttl: 0, // No Cache
+    },
+    async () => {
+      try {
+        const result = await query<AccountId[]>(getAccIdByUserNameQuery, [userName]);
+        if (result[0] === undefined) {
+          return -1;
+        }
+        return result[0].accId;
+      } catch (error) {
+        console.log(`[getAccountIdByUserName error]: `, error);
+      }
+    }
+  );
+};
+
+/**
+ * Bans Account by Account Id
+ *
+ * @param {string} userName  The user name / login of the account
+ * @returns {string} Result Message
+ */
+export const banPlayerByAccId = async (userName: string) => {
+  return cache.get(
+    {
+      key: `banPlayerByAccId_${userName}`,
+      ttl: 0, // No Cache
+    },
+    async () => {
+      try {
+        const accId = await getAccountIdByUserName(userName);
+        if (accId === -1) {
+          return `No Account Id Found`;
```

Comment:
thought (non-blocking): as a user of this function, it may be hard to intepret the results as I would have to do string comparisons. I wonder if we should establish a response pattern in the form of something like:
```
return {
  error: string | undefined,
  data: any
}
```
So users of this function can quickly parse if the return value has an error and display it, and if not handle the data.

Something I've seen in a library recently:
`  const { data, error, isLoading } = useSWR('/api/user', fetcher)`
https://swr.vercel.app/#overview